$MirOS: ports/lang/python/2.5/patches/patch-Objects_fileobject_c,v 1.1 2008/08/05 19:11:50 bsiegert Exp $
--- Objects/fileobject.c.orig	Tue Dec 23 13:10:20 2008
+++ Objects/fileobject.c	Sat May 30 19:02:27 2009
@@ -831,7 +831,6 @@ file_read(PyFileObject *f, PyObject *arg
 	long bytesrequested = -1;
 	size_t bytesread, buffersize, chunksize;
 	PyObject *v;
-
 	if (f->f_fp == NULL)
 		return err_closed();
 	/* refuse to mix with f.next() */
@@ -845,6 +844,7 @@ file_read(PyFileObject *f, PyObject *arg
 		buffersize = new_buffersize(f, (size_t)0);
 	else
 		buffersize = bytesrequested;
+
 	if (buffersize > PY_SSIZE_T_MAX) {
 		PyErr_SetString(PyExc_OverflowError,
 	"requested number of bytes is more than a Python string can hold");
@@ -1075,7 +1075,7 @@ getline_via_fgets(FILE *fp)
 	 * into its buffer.
 	 */
 	total_v_size = MAXBUFSIZE << 1;
-	v = PyString_FromStringAndSize((char*)NULL, (int)total_v_size);
+	v = PyString_FromStringAndSize((char*)NULL, total_v_size);
 	if (v == NULL)
 		return v;
 	/* copy over everything except the last null byte */
@@ -1130,7 +1130,7 @@ getline_via_fgets(FILE *fp)
 			Py_DECREF(v);
 			return NULL;
 		}
-		if (_PyString_Resize(&v, (int)total_v_size) < 0)
+		if (_PyString_Resize(&v, (Py_ssize_t)total_v_size) < 0)
 			return NULL;
 		/* overwrite the trailing null byte */
 		pvfree = BUF(v) + (prev_v_size - 1);
